core/src/test/java/org/apache/accumulo/core/rpc/AccumuloProtocolTest.java

```diff
@@ -103,6 +105,59 @@ public void testIncompatibleVersion() throws TException {
     }
   }
 
+  /**
+   * Test that compatible accumulo version (same major.minor) passes validation
+   */
+  @Test
+  public void testCompatibleVersions() throws TException {
+    try (TMemoryBuffer transport = new TMemoryBuffer(100)) {
+      TCompactProtocol protocol = new TCompactProtocol(transport);
+      protocol.writeI32(VALID_MAGIC_NUMBER);
+      protocol.writeByte(VALID_PROTOCOL_VERSION);
+
+      // Write current version but with different patch version
+      String serverMajorMinor = Constants.VERSION.substring(0, Constants.VERSION.lastIndexOf('.'));
+      String clientVersion = serverMajorMinor + ".999";
+
+      protocol.writeString(clientVersion);
+      protocol.writeBool(false);
+
+      AccumuloProtocolFactory.AccumuloProtocol serverProtocol =
+          (AccumuloProtocolFactory.AccumuloProtocol) AccumuloProtocolFactory.serverFactory()
+              .getProtocol(transport);
+
+      assertDoesNotThrow(serverProtocol::validateHeader,
+          "Expected compatible version to pass validation");
```

Comment:
I think assertDoesNotThrow can be useful when we want to ignore a return value that normally would be thrown and expected to be used by some static analysis tool (like spotbugs). But in most code, we can just skip it altogether, as in:

```suggestion
      serverProtocol.validateHeader();
```